### PR TITLE
[CI:BUILD] Build universal Podman binary for Mac installer

### PR DESCRIPTION
**Pull request by ashley-cui**
Build universal Podman binary and installer for Mac. Update GitHub action to build it too.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ashley-cui** [comment]
@containers/podman-maintainers Would also like to get this in before today's RC, PTAL

**Comment by ashley-cui** [comment]
Fixes: https://github.com/containers/podman/issues/21829, https://github.com/containers/podman/issues/21769

File: contrib/pkginstaller/Makefile [hunk 1]
```diff
@@ -11,22 +11,22 @@ VFKIT_VERSION ?= 0.5.1
...
-TMP_DOWNLOAD ?= tmp-download
+TMP_BIN ?= tmp-bin
 PACKAGE_ROOT ?= root
 PKG_NAME := podman-installer-macos-$(GOARCH).pkg
 
 default: pkginstaller
 
 podman_version:
-	make -C ../../ test/version/version
+	make -B -C ../../ test/version/version
```
Comment (TomSweeneyRedHat) [inline contrib/pkginstaller/Makefile:21]: I'd prefer:

```suggestion
	make --always-make --directory ../../ test/version/version
```

But `¯\_(ツ)_/¯`

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Review by Luap99 — COMMENTED** [review]
Does this correctly include universal gvproxy/vfkit binaries as well? Just asking because we seem to just download whatever they have on their github repo. I am assume you checked this but asking anyway in case they do not provide "universal" binaries there.

**Comment by ashley-cui** [comment]
@Luap99 Both are universal! 
```
gvproxy:           Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
gvproxy (for architecture x86_64):	Mach-O 64-bit executable x86_64
gvproxy (for architecture arm64):	Mach-O 64-bit executable arm64
vfkit:             Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
vfkit (for architecture x86_64):	Mach-O 64-bit executable x86_64
vfkit (for architecture arm64):	Mach-O 64-bit executable arm64
```
And the new Podman binaries with this PR:
```
podman:            Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
podman (for architecture x86_64):	Mach-O 64-bit executable x86_64
podman (for architecture arm64):	Mach-O 64-bit executable arm64
podman-mac-helper: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
podman-mac-helper (for architecture x86_64):	Mach-O 64-bit executable x86_64
podman-mac-helper (for architecture arm64):	Mach-O 64-bit executable arm64
```

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**